app/main.py

```diff
@@ -2,24 +2,76 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: str, name: str) -> None:
```

Comment:
The `__set_name__` method has incorrect type annotations. The `owner` parameter should be of type `type`, not `str`, and `name` should be `str` as it is. Update the type annotations to reflect this.